### PR TITLE
School Library: Preserve Data

### PR DESCRIPTION
**Pull request by IndieCoderMM**
# Storing and loading app data
We implemented the following features in this branch:
- Added methods to convert objects to hashes
- Stored books, people, and rentals into JSON files
- Created `Storage` class for handling data

### CODE REVIEW

File: storage.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  private
+
+  def read_data(filename)
+    file = File.open(filename)
+    file_data = file.read
+    file.close
+    return JSON.parse(file_data) unless file_data.empty?
+
+    []
```
Comment (OyePriscilla) [inline storage.rb:34]: - [ ] Please kindly adjust this logic, according to the requirement when one or more of the files are missing the app should not throw any error. I got this error when I deleted rentals.json file. Perhaps, this `return [] unless File.exists?("filename.json")` will resolve the issue. 


Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @IndieCoderMM, @yetemegn-telaye and @Ridwanullahi-code 👍  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Preserve the data from the app ✔️ 
Load preserved data on startup ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: storage.rb (hunk 1 above)
Comment (IndieCoderMM) [reply storage.rb:34]: Thanks for your suggestion. I'll implement it right away.

**Review by OyePriscilla — APPROVED** [review]
Hi @IndieCoderMM, @yetemegn-telaye and @Ridwanullahi-code 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Preserve the data from the app ✔️
App worked as expected ✔️ 
Made requires changes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**